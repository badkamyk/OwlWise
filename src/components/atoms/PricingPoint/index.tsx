export default function PricingPoint({ text }: { text: string }) {
	return (
		<li className="flex items-center space-x-3">
			<svg
				className="h-5 w-5 flex-shrink-0 text-green-500 dark:text-green-400"
				fill="currentColor"
				viewBox="0 0 20 20"
				xmlns="http://www.w3.org/2000/svg"
			>
				<path
					fillRule="evenodd"
					d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
					clipRule="evenodd"
				></path>
			</svg>
			<span className={"text-white"}>{text}</span>
		</li>
	);
}
