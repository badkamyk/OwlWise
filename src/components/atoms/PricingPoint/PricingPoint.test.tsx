import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import PricingPoint from "./index";

it("renders PricingPoint with props correctly", () => {
	const { container } = render(<PricingPoint text="PricingPoint" />);
	expect(
		container.getElementsByClassName("flex items-center space-x-3")
	).toBeTruthy();

	expect(
		container.getElementsByClassName(
			"h-5 w-5 flex-shrink-0 text-green-500 dark:text-green-400"
		)
	).toBeTruthy();

	expect(screen.getByText("PricingPoint")).toBeTruthy();
	expect(container).toMatchSnapshot();
});
