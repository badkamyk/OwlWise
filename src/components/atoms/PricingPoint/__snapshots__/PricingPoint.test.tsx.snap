// Vitest Snapshot v1

exports[`PricingPoint with props 1`] = `
<div>
  <li
    class="flex items-center space-x-3"
  >
    <svg
      class="h-5 w-5 flex-shrink-0 text-green-500 dark:text-green-400"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
        fill-rule="evenodd"
      />
    </svg>
    <span
      class="text-white"
    >
      PricingPoint
    </span>
  </li>
</div>
`;
